### ISSUE TITLE
[28] Emerging Properties in Self-Supervised Vision Transformers (DINO)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#56]
[paper](https://arxiv.org/pdf/2104.14294.pdf)
[code](https://github.com/facebookresearch/dino)
[yannic youtube](https://www.youtube.com/watch?v=h3ij3F3cPIk&t=925s)

self-supervised 잘 해서 [CLS] token의 last-layer self-attention 뽑아보면 unsupervised segmentation 이 되더라.
![image](https://user-images.githubusercontent.com/16400591/141235560-49359b75-78f9-41ab-a77a-2aefd388e018.png)

BYOL 과 유사하지만, 새로운 방식의 self-supervised learning 기법 제안

# DINO
방법은 정말정말 단순하다.
하나의 이미지에 대해 다른 augmentation을 적용하고, teacher model 과 student model 에 통과 시킨다.
총 4개의 output 이 나오게 되고 이들을 이용해 loss 를 구한다.
backward 는 student 에 대해서만 수행하고, teacher model 은 student model weight 의 ema 를 이용해 업데이트 시킨다.
centering 과 sharpening 에 대한 설명은 조금 더 아래에서 설명한다.

![image](https://user-images.githubusercontent.com/16400591/141930816-19cf1ad1-5511-4082-97b6-d35fbfe38cfe.png)
![image](https://user-images.githubusercontent.com/16400591/141930843-6489b83c-a051-4165-84ae-c9922cc66965.png)

## Centering & Sharpening
DINO 는 2가지 mode collapse 가 존재한다.
1. input에 상관 없이 uniform logit 이 나오는 것.
2. class 하나에 대해서만 logit 이 크게 나오는 것.

(2) 를 해결하기 위해 centering 이 들어가고, (1) 을 해결하기 위해 sharpening 이 들어간다.
C 는 teacher output logit 들을 평균내서 업데이트 시키게 된다.
그 다음, loss 를 구할 때 C logit을 빼주는 식으로 작동한다. (위 알고리즘 참고)

collapse 방지에 얼마나 효과가 있는지 보기 위한 ablation graph.
teacher 와 student 의 output 이 똑같지 않도록 잘 만들어 준다.
![image](https://user-images.githubusercontent.com/16400591/141933265-986dadfc-9e28-429f-90d1-62c1481966b9.png)

## Results
![image](https://user-images.githubusercontent.com/16400591/141934508-a30b181d-632a-4c0d-9b92-936a9945893b.png)
이외에도 다양한 set 에 대해 비교를 진행했다. 
굳이 스크랩 하지는 않았다.

### Ablations
ablation 이 재밌다. VIT 기준으로 어떤 component가 큰 영향을 끼칠까 조사를 해 보았다.
![image](https://user-images.githubusercontent.com/16400591/141934892-05eea12c-ac5d-4498-84f5-ebe264d35a5e.png)
multicrop 을 안 쓰면, 4% 정도 하락
**주목할 만한 점은 MSE loss 를 쓰면 14% 가 떨어진다는 점.**
~개인적으로는 softmax 가 logit 의 scale normalize 를 해주다 보니 학습에 좋은 영향을 끼친 것 같다 생각함~

### Multi-crop
SwAV 에서 사용하던 방법. 
기본 view 는 2개 (2개의 다른 augmentation) 이다. 추가로 10개의 smaller view 들을 crop 해서 구하게 되는데, 이 crop 된 작은 이미지들은 모두 **student 만 통과**시키게 된다.
그리고 모든 pair 에 대해 loss 를 계산해 주면 된다. 간단하다.
![image](https://user-images.githubusercontent.com/16400591/141941656-dcf02853-78eb-4990-8ac1-d7fcf270cac0.png)

Multi-crop 세팅에 대한 ablation study
![image](https://user-images.githubusercontent.com/16400591/141936450-1e88b1b2-84e2-4f5e-b9da-4ac7651078b1.png)
